src/lib.rs

```diff
@@ -79,11 +79,22 @@ impl Authorizer {
             );
 
             // Verify that the capability is actually reached through:
-            ensure!(
-                proof.payload.capability_key() == &self.identity,
-                "invocation failed: proof is missing delegation for capability of {}",
-                hex::encode(self.identity)
-            );
+            match proof.payload.capability_origin() {
```

Comment:
I think it's simpler if you provide a function `Payload::capability_issuer` which returns `payload.issuer` if `CapabilityOrigin` is `Issuer`, but otherwise returns the `VerifyingKey` inside `Delegation`.